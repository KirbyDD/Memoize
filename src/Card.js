import React, { Component } from "react";
import "./Card.scss"

class Card extends Component {
  constructor(props){
    super(props);
    this.state = {
      answerInput: "",
      resultMsg: null
    }
  }

  captureChange = (e) => {
    this.setState({ answerInput: e.target.value.toLowerCase()});
  }

  handleAnswer = (e) => {
    e.preventDefault();
    if(this.state.answerInput === this.props.answer.toLowerCase()){
      this.setState({ resultMsg: "Trueeeeeee!"})
    } else {
      this.setState({ resultMsg: "Nahh try again!"})
    }
  }

  render(){
    let problemImg;
    if(this.props.image){
      problemImg = <img src={this.props.image} className="problem-img"/>;
    }
    return(
      <article>
        <h2>Question: {this.props.key}</h2>
        {problemImg}
        <h3>{this.props.question}</h3>
        {this.state.resultMsg}
        <form>
          <input type="text" placeholder=" Type answer here..." onChange={this.captureChange}/>
          <div>
            <button className="answer-button" onClick={this.handleAnswer}>Check Answer</button>
            <button className="prev-button" onClick={this.props.handlePrev}>Previous Question</button>
            <button className="next-button" onClick={this.props.handleNext}>Next Question</button>
          </div>
        </form>
      </article>
    )
  }
}

export default Card;