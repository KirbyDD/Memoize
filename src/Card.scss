article{
  align-items: center;
  background-color: white;
  box-shadow: 5px 5px 10px;
  display: flex;
  flex-direction: column;
  width: 80%;
}

.problem-img {
  width: 600px;
}

form {
  align-items: center;
  display: flex;
  flex-direction: column;
  input {
    font-size: 1rem;
    height: 30px;
    margin: 10px auto 30px;
    width: 300px;
  }
}

button {
  font-size: 15px;
  font-weight: bold;
  height: 37px;
  width: 200px;
}

