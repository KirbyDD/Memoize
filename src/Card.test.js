import React from 'react';
import Card from './Card.js';
import { shallow } from 'enzyme';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() });

const mockQuestion = {
  id: 1,
  question: "Which of the following is an ES5 variable declaration: var, let, or const?",
  image: true,
  answer: "var"
  }

const handleNext = jest.fn();
const handlePrev = jest.fn();
const handleAnswer = jest.fn();


describe('CardContainer', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <Card handleNext={handleNext}
            handlePrev={handlePrev}
            question={mockQuestion.question}
            answer={mockQuestion.answer}
            key={mockQuestion.id}
            image={mockQuestion.image}/>
    )
  })

  it('should match snapshot when data is passed in ', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should have state with default value', () => {
    expect(wrapper.state()).toEqual({ answerInput: "", resultMsg: null });
  });
  it('should call handleAnswer on click of answer button', () => {
    wrapper.setState({  answerInput: 'var'});
    wrapper.find(".answer-button").simulate('click', { preventDefault:() =>{}});
    expect(wrapper.state('resultMsg')).toEqual("Trueeeeeee!")
  });
  it('should call handleAnswer on click of answer button', () => {
    wrapper.setState({  answerInput: 'let'});
    wrapper.find(".answer-button").simulate('click', { preventDefault:() =>{}});
    expect(wrapper.state('resultMsg')).toEqual("Nahh try again!")
  });
  it('should update the state of the answerInput on change', () =>{
    expect(wrapper.state('answerInput')).toEqual("")
    wrapper.instance().captureChange({ target: { value: "var"}} )
    expect(wrapper.state('answerInput')).toEqual("var")
  });
})