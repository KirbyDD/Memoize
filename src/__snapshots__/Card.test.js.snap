// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match snapshot when data is passed in  1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    addStudyCard={[MockFunction]}
    answer="var"
    handleNext={[MockFunction]}
    handlePrev={[MockFunction]}
    id={1}
    image={true}
    question="Which of the following is an ES5 variable declaration: var, let, or const?"
    studyList={
      Array [
        1,
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2>
          Question: 
          1
        </h2>,
        <img
          className="problem-img"
          src={true}
        />,
        <h3>
          Which of the following is an ES5 variable declaration: var, let, or const?
        </h3>,
        null,
        <form>
          <input
            onChange={[Function]}
            placeholder=" Type answer here..."
            type="text"
          />
          <div>
            <button
              className="answer-button"
              onClick={[Function]}
            >
              Check Answer
            </button>
            <button
              className="study-list-button"
              onClick={[Function]}
            >
              Remove from Study List 
            </button>
            <button
              className="prev-button"
              onClick={[MockFunction]}
            >
              Previous Question
            </button>
            <button
              className="next-button"
              onClick={[MockFunction]}
            >
              Next Question
            </button>
          </div>
        </form>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Question: ",
            1,
          ],
        },
        "ref": null,
        "rendered": Array [
          "Question: ",
          1,
        ],
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "problem-img",
          "src": true,
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Which of the following is an ES5 variable declaration: var, let, or const?",
        },
        "ref": null,
        "rendered": "Which of the following is an ES5 variable declaration: var, let, or const?",
        "type": "h3",
      },
      null,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <input
              onChange={[Function]}
              placeholder=" Type answer here..."
              type="text"
            />,
            <div>
              <button
                className="answer-button"
                onClick={[Function]}
              >
                Check Answer
              </button>
              <button
                className="study-list-button"
                onClick={[Function]}
              >
                Remove from Study List 
              </button>
              <button
                className="prev-button"
                onClick={[MockFunction]}
              >
                Previous Question
              </button>
              <button
                className="next-button"
                onClick={[MockFunction]}
              >
                Next Question
              </button>
            </div>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "onChange": [Function],
              "placeholder": " Type answer here...",
              "type": "text",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <button
                  className="answer-button"
                  onClick={[Function]}
                >
                  Check Answer
                </button>,
                <button
                  className="study-list-button"
                  onClick={[Function]}
                >
                  Remove from Study List 
                </button>,
                <button
                  className="prev-button"
                  onClick={[MockFunction]}
                >
                  Previous Question
                </button>,
                <button
                  className="next-button"
                  onClick={[MockFunction]}
                >
                  Next Question
                </button>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Check Answer",
                  "className": "answer-button",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "Check Answer",
                "type": "button",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Remove from Study List ",
                  "className": "study-list-button",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "Remove from Study List ",
                "type": "button",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Previous Question",
                  "className": "prev-button",
                  "onClick": [MockFunction],
                },
                "ref": null,
                "rendered": "Previous Question",
                "type": "button",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Next Question",
                  "className": "next-button",
                  "onClick": [MockFunction],
                },
                "ref": null,
                "rendered": "Next Question",
                "type": "button",
              },
            ],
            "type": "div",
          },
        ],
        "type": "form",
      },
    ],
    "type": "article",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2>
            Question: 
            1
          </h2>,
          <img
            className="problem-img"
            src={true}
          />,
          <h3>
            Which of the following is an ES5 variable declaration: var, let, or const?
          </h3>,
          null,
          <form>
            <input
              onChange={[Function]}
              placeholder=" Type answer here..."
              type="text"
            />
            <div>
              <button
                className="answer-button"
                onClick={[Function]}
              >
                Check Answer
              </button>
              <button
                className="study-list-button"
                onClick={[Function]}
              >
                Remove from Study List 
              </button>
              <button
                className="prev-button"
                onClick={[MockFunction]}
              >
                Previous Question
              </button>
              <button
                className="next-button"
                onClick={[MockFunction]}
              >
                Next Question
              </button>
            </div>
          </form>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Question: ",
              1,
            ],
          },
          "ref": null,
          "rendered": Array [
            "Question: ",
            1,
          ],
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "problem-img",
            "src": true,
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Which of the following is an ES5 variable declaration: var, let, or const?",
          },
          "ref": null,
          "rendered": "Which of the following is an ES5 variable declaration: var, let, or const?",
          "type": "h3",
        },
        null,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <input
                onChange={[Function]}
                placeholder=" Type answer here..."
                type="text"
              />,
              <div>
                <button
                  className="answer-button"
                  onClick={[Function]}
                >
                  Check Answer
                </button>
                <button
                  className="study-list-button"
                  onClick={[Function]}
                >
                  Remove from Study List 
                </button>
                <button
                  className="prev-button"
                  onClick={[MockFunction]}
                >
                  Previous Question
                </button>
                <button
                  className="next-button"
                  onClick={[MockFunction]}
                >
                  Next Question
                </button>
              </div>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "onChange": [Function],
                "placeholder": " Type answer here...",
                "type": "text",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <button
                    className="answer-button"
                    onClick={[Function]}
                  >
                    Check Answer
                  </button>,
                  <button
                    className="study-list-button"
                    onClick={[Function]}
                  >
                    Remove from Study List 
                  </button>,
                  <button
                    className="prev-button"
                    onClick={[MockFunction]}
                  >
                    Previous Question
                  </button>,
                  <button
                    className="next-button"
                    onClick={[MockFunction]}
                  >
                    Next Question
                  </button>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Check Answer",
                    "className": "answer-button",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": "Check Answer",
                  "type": "button",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Remove from Study List ",
                    "className": "study-list-button",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": "Remove from Study List ",
                  "type": "button",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Previous Question",
                    "className": "prev-button",
                    "onClick": [MockFunction],
                  },
                  "ref": null,
                  "rendered": "Previous Question",
                  "type": "button",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Next Question",
                    "className": "next-button",
                    "onClick": [MockFunction],
                  },
                  "ref": null,
                  "rendered": "Next Question",
                  "type": "button",
                },
              ],
              "type": "div",
            },
          ],
          "type": "form",
        },
      ],
      "type": "article",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;

exports[`Card should match snapshot when data is passed in  2`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    addStudyCard={[MockFunction]}
    answer="var"
    handleNext={[MockFunction]}
    handlePrev={[MockFunction]}
    id={1}
    image={true}
    question="Which of the following is an ES5 variable declaration: var, let, or const?"
    studyList={
      Array [
        1,
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2>
          Question: 
          1
        </h2>,
        <img
          className="problem-img"
          src={true}
        />,
        <h3>
          Which of the following is an ES5 variable declaration: var, let, or const?
        </h3>,
        null,
        <form>
          <input
            onChange={[Function]}
            placeholder=" Type answer here..."
            type="text"
          />
          <div>
            <button
              className="answer-button"
              onClick={[Function]}
            >
              Check Answer
            </button>
            <button
              className="study-list-button"
              onClick={[Function]}
            >
              Remove from Study List 
            </button>
            <button
              className="prev-button"
              onClick={[MockFunction]}
            >
              Previous Question
            </button>
            <button
              className="next-button"
              onClick={[MockFunction]}
            >
              Next Question
            </button>
          </div>
        </form>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Question: ",
            1,
          ],
        },
        "ref": null,
        "rendered": Array [
          "Question: ",
          1,
        ],
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "problem-img",
          "src": true,
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Which of the following is an ES5 variable declaration: var, let, or const?",
        },
        "ref": null,
        "rendered": "Which of the following is an ES5 variable declaration: var, let, or const?",
        "type": "h3",
      },
      null,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <input
              onChange={[Function]}
              placeholder=" Type answer here..."
              type="text"
            />,
            <div>
              <button
                className="answer-button"
                onClick={[Function]}
              >
                Check Answer
              </button>
              <button
                className="study-list-button"
                onClick={[Function]}
              >
                Remove from Study List 
              </button>
              <button
                className="prev-button"
                onClick={[MockFunction]}
              >
                Previous Question
              </button>
              <button
                className="next-button"
                onClick={[MockFunction]}
              >
                Next Question
              </button>
            </div>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "onChange": [Function],
              "placeholder": " Type answer here...",
              "type": "text",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <button
                  className="answer-button"
                  onClick={[Function]}
                >
                  Check Answer
                </button>,
                <button
                  className="study-list-button"
                  onClick={[Function]}
                >
                  Remove from Study List 
                </button>,
                <button
                  className="prev-button"
                  onClick={[MockFunction]}
                >
                  Previous Question
                </button>,
                <button
                  className="next-button"
                  onClick={[MockFunction]}
                >
                  Next Question
                </button>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Check Answer",
                  "className": "answer-button",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "Check Answer",
                "type": "button",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Remove from Study List ",
                  "className": "study-list-button",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "Remove from Study List ",
                "type": "button",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Previous Question",
                  "className": "prev-button",
                  "onClick": [MockFunction],
                },
                "ref": null,
                "rendered": "Previous Question",
                "type": "button",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Next Question",
                  "className": "next-button",
                  "onClick": [MockFunction],
                },
                "ref": null,
                "rendered": "Next Question",
                "type": "button",
              },
            ],
            "type": "div",
          },
        ],
        "type": "form",
      },
    ],
    "type": "article",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2>
            Question: 
            1
          </h2>,
          <img
            className="problem-img"
            src={true}
          />,
          <h3>
            Which of the following is an ES5 variable declaration: var, let, or const?
          </h3>,
          null,
          <form>
            <input
              onChange={[Function]}
              placeholder=" Type answer here..."
              type="text"
            />
            <div>
              <button
                className="answer-button"
                onClick={[Function]}
              >
                Check Answer
              </button>
              <button
                className="study-list-button"
                onClick={[Function]}
              >
                Remove from Study List 
              </button>
              <button
                className="prev-button"
                onClick={[MockFunction]}
              >
                Previous Question
              </button>
              <button
                className="next-button"
                onClick={[MockFunction]}
              >
                Next Question
              </button>
            </div>
          </form>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Question: ",
              1,
            ],
          },
          "ref": null,
          "rendered": Array [
            "Question: ",
            1,
          ],
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "problem-img",
            "src": true,
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Which of the following is an ES5 variable declaration: var, let, or const?",
          },
          "ref": null,
          "rendered": "Which of the following is an ES5 variable declaration: var, let, or const?",
          "type": "h3",
        },
        null,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <input
                onChange={[Function]}
                placeholder=" Type answer here..."
                type="text"
              />,
              <div>
                <button
                  className="answer-button"
                  onClick={[Function]}
                >
                  Check Answer
                </button>
                <button
                  className="study-list-button"
                  onClick={[Function]}
                >
                  Remove from Study List 
                </button>
                <button
                  className="prev-button"
                  onClick={[MockFunction]}
                >
                  Previous Question
                </button>
                <button
                  className="next-button"
                  onClick={[MockFunction]}
                >
                  Next Question
                </button>
              </div>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "onChange": [Function],
                "placeholder": " Type answer here...",
                "type": "text",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <button
                    className="answer-button"
                    onClick={[Function]}
                  >
                    Check Answer
                  </button>,
                  <button
                    className="study-list-button"
                    onClick={[Function]}
                  >
                    Remove from Study List 
                  </button>,
                  <button
                    className="prev-button"
                    onClick={[MockFunction]}
                  >
                    Previous Question
                  </button>,
                  <button
                    className="next-button"
                    onClick={[MockFunction]}
                  >
                    Next Question
                  </button>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Check Answer",
                    "className": "answer-button",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": "Check Answer",
                  "type": "button",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Remove from Study List ",
                    "className": "study-list-button",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": "Remove from Study List ",
                  "type": "button",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Previous Question",
                    "className": "prev-button",
                    "onClick": [MockFunction],
                  },
                  "ref": null,
                  "rendered": "Previous Question",
                  "type": "button",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Next Question",
                    "className": "next-button",
                    "onClick": [MockFunction],
                  },
                  "ref": null,
                  "rendered": "Next Question",
                  "type": "button",
                },
              ],
              "type": "div",
            },
          ],
          "type": "form",
        },
      ],
      "type": "article",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
